src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -178,13 +164,85 @@ public LocalCall(String functionName, Optional<List<?>> arg,
             final SaltClient client, Target<?> target,
             String username, String password, AuthModule authModule)
             throws SaltException {
+        return callSyncBatchable(client, target, Optional.of(username),
+                Optional.of(password), Optional.of(authModule), Optional.empty()).get(0);
+    }
+
+    /**
+     * Calls a execution module function on the given target with batching and
+     * synchronously waits for the result. Authentication is done with the token
+     * therefore you have to login prior to using this function.
+     *
+     * @param client SaltClient instance
+     * @param target the target for the function
+     * @param batch  the batch parameter, e.g. 10, 10%, etc.
+     * @return A list of maps with each list representing each batch, and maps containing
+     * the results with the minion names as keys.
+     * @throws SaltException if anything goes wrong
+     */
+    public List<Map<String, Result<R>>> callSyncBatch(
```

Comment:
Hi, thanks for the feedback. I saw that you had mentioned simply overloading the `callSync` methods in a previous comment, but I noticed that the other values for the `Client` enum seemed to have separate methods as well, so I followed the pattern, e.g. `Client.LOCAL` has `callSync`, `Client.LOCAL_ASYNC` has `callAsync`, `Client.SSH` has `callSyncSSH`.